portal/session.py

```diff
@@ -0,0 +1,84 @@
+import os
+
+from flask_session import Session, RedisSessionInterface
+from flask_session.sessions import total_seconds
+
+
+class BaseRedisSessionInterface(RedisSessionInterface):
+    """Override RedisSessionInterface save_session() method to passthrough flask config"""
+
+    def save_session(self, app, session, response):
+        domain = self.get_cookie_domain(app)
+        path = self.get_cookie_path(app)
+        if not session:
+            if session.modified:
+                self.redis.delete(self.key_prefix + session.sid)
+                response.delete_cookie(
+                    app.session_cookie_name,
+                    domain=domain,
+                    path=path,
+                )
+            return
+
+        # Modification case.  There are upsides and downsides to
+        # emitting a set-cookie header each request.  The behavior
+        # is controlled by the :meth:`should_set_cookie` method
+        # which performs a quick check to figure out if the cookie
+        # should be set or not.  This is controlled by the
+        # SESSION_REFRESH_EACH_REQUEST config flag as well as
+        # the permanent flag on the session itself.
+        # if not self.should_set_cookie(app, session):
+        #    return
+
+        httponly = self.get_cookie_httponly(app)
+        secure = self.get_cookie_secure(app)
+        # call new flask config method for reading configured sameSite cookie value
+        samesite = self.get_cookie_samesite(app)
+        expires = self.get_expiration_time(app, session)
+        val = self.serializer.dumps(dict(session))
+        self.redis.setex(
+            name=self.key_prefix + session.sid, value=val,
+            time=total_seconds(app.permanent_session_lifetime),
+        )
+        if self.use_signer:
+            session_id = self._get_signer(app).sign(want_bytes(session.sid))
+        else:
+            session_id = session.sid
+
+        response.set_cookie(
+            app.session_cookie_name,
+            session_id,
+            expires=expires,
+            httponly=httponly,
+            domain=domain,
+            path=path,
+            secure=secure,
+            samesite=samesite,
+        )
+
+
+class SameSiteSession(Session):
+    """Extends flask-session to passthrough SESSION_COOKIE_SAMESITE flask config"""
+    def _get_interface(self, app):
+        config = app.config.copy()
+        config.setdefault('SESSION_TYPE', 'null')
+        config.setdefault('SESSION_PERMANENT', True)
+        config.setdefault('SESSION_USE_SIGNER', False)
+        config.setdefault('SESSION_KEY_PREFIX', 'session:')
+        config.setdefault('SESSION_REDIS', None)
+        config.setdefault('SESSION_MEMCACHED', None)
+        config.setdefault('SESSION_FILE_DIR', os.path.join(os.getcwd(), 'flask_session'))
+        config.setdefault('SESSION_FILE_THRESHOLD', 500)
+        config.setdefault('SESSION_FILE_MODE', 384)
+        config.setdefault('SESSION_MONGODB', None)
```

Comment:
Looks like all permutations of SESSION storage are covered.  Is this just in case we adopt a different session type, or because it's also done in the base class, or ?

My concern is maintenance - I'd hope we can just inherit the defaults, surprised we'd need to call out so many?

Reply:
>Looks like all permutations of SESSION storage are covered. Is this just in case we adopt a different session type, or because it's also done in the base class, or ?

Yup, I just copied too much from the [base class](https://github.com/fengsp/flask-session/blob/0.3.2/flask_session/__init__.py#L63-L109), when we only really care about supporting redis. I'll go ahead and remove references to other server-side session backends